lib/restrictedsession/restricted_test.go

```diff
@@ -242,198 +210,128 @@ func (s *Suite) SetUpSuite(c *check.C) {
+func (tt *bpfContext) dialExpectDeny(t *testing.T, ver int, ip string) {
 	// Only EPERM is expected
 	err := dialTCP(ver, mustParseIP(ip))
 	if err == nil {
-		c.Fatalf("Dial %v: did not expect to succeed", ip)
+		t.Fatalf("Dial %v: did not expect to succeed", ip)
 	}
 
 	if !errors.Is(err, syscall.EPERM) {
-		c.Fatalf("Dial %v: EPERM expected, got: %v", ip, err)
+		t.Fatalf("Dial %v: EPERM expected, got: %v", ip, err)
 	}
 
-	ev := s.expectedAuditEvent(ver, ip, apievents.SessionNetwork_CONNECT)
-	s.expectedAuditEvents = append(s.expectedAuditEvents, ev)
+	ev := tt.expectedAuditEvent(ver, ip, apievents.SessionNetwork_CONNECT)
+	tt.expectedAuditEvents = append(tt.expectedAuditEvents, ev)
 }
 
-func (s *Suite) sendExpectAllow(c *check.C, ver int, ip string) {
-	err := sendUDP(ver, mustParseIP(ip))
-
-	// Other than EPERM or EINVAL is OK
-	if errors.Is(err, syscall.EPERM) || errors.Is(err, syscall.EINVAL) {
-		c.Fatalf("Send %v: failed with %v", ip, err)
-	}
-}
-
-func (s *Suite) sendExpectDeny(c *check.C, ver int, ip string) {
+func (tt *bpfContext) sendExpectDeny(t *testing.T, ver int, ip string) {
 	err := sendUDP(ver, mustParseIP(ip))
 	if !errors.Is(err, syscall.EPERM) {
-		c.Fatalf("Send %v: was not denied: %v", ip, err)
+		t.Fatalf("Send %v: was not denied: %v", ip, err)
 	}
 
-	ev := s.expectedAuditEvent(ver, ip, apievents.SessionNetwork_SEND)
-	s.expectedAuditEvents = append(s.expectedAuditEvents, ev)
+	ev := tt.expectedAuditEvent(ver, ip, apievents.SessionNetwork_SEND)
+	tt.expectedAuditEvents = append(tt.expectedAuditEvents, ev)
 }
 
-func (s *Suite) expectedAuditEvent(ver int, ip string, op apievents.SessionNetwork_NetworkOperation) apievents.AuditEvent {
+func (tt *bpfContext) expectedAuditEvent(ver int, ip string, op apievents.SessionNetwork_NetworkOperation) apievents.AuditEvent {
 	return &apievents.SessionNetwork{
 		Metadata: apievents.Metadata{
 			Type: events.SessionNetworkEvent,
 			Code: events.SessionNetworkCode,
 		},
 		ServerMetadata: apievents.ServerMetadata{
-			ServerID:        s.ctx.ServerID,
-			ServerNamespace: s.ctx.Namespace,
+			ServerID:        tt.ctx.ServerID,
+			ServerNamespace: tt.ctx.Namespace,
 		},
 		SessionMetadata: apievents.SessionMetadata{
-			SessionID: s.ctx.SessionID,
+			SessionID: tt.ctx.SessionID,
 		},
 		UserMetadata: apievents.UserMetadata{
-			User:  s.ctx.User,
-			Login: s.ctx.Login,
+			User:  tt.ctx.User,
+			Login: tt.ctx.Login,
 		},
 		BPFMetadata: apievents.BPFMetadata{
-			CgroupID: s.cgroupID,
+			CgroupID: tt.cgroupID,
 			Program:  "restrictedsessi",
-			PID:      uint64(s.ctx.PID),
+			PID:      uint64(tt.ctx.PID),
 		},
 		DstPort:    testPort,
 		DstAddr:    ip,
-		SrcAddr:    s.srcAddrs[ver],
+		SrcAddr:    tt.srcAddrs[ver],
 		TCPVersion: int32(ver),
 		Operation:  op,
 		Action:     apievents.EventAction_DENIED,
 	}
 }
 
-func (s *Suite) TestNetwork(c *check.C) {
+func TestNetwork(t *testing.T) {
+	tt := setupBPFContext(t)
+	t.Cleanup(func() { tt.Close(t) })
```

Comment:
I would move the cleanup call into `setupBPFContext`.

I've been encouraging the guideline of "if the setup func takes a testing.T, then it should register all cleanup"